prow/cluster/starter.yaml

```diff
@@ -356,7 +354,7 @@ spec:
   rules:
   - http:
       paths:
-      - path: /*
+      - path: /
```

Comment:
With this match deck paths like https://prow.k8s.io/pr and https://prow.k8s.io/badge.svg?

Reply:
Yes. Ingress paths are matched in a "everything that starts with this path" and "narrowest match first" fashion.

Since a literal `*` may be part of an URI (TIL), I would consider an ingress-controller implementation that interprets this as some wildcard specifier as faulty :)

Reply:
This got me on GKE's ingress, and I had to revert this to make other paths work.

The guide is using GKE by default currently. Should we revert this change or file an issue in https://github.com/kubernetes/ingress-gce?

Reply:
cc @MrHohn @rramkumar1  :-)

Reply:
@idealhack Very interesting finding, thanks for sharing. Regarding the `The guide is using GKE by default currently` we do also use GKE but not the GCP ingress controller, the former does not necessarily imply the latter.

Looking up the [godocs](https://github.com/kubernetes/kubernetes/blob/f83bb59047b1de54dd24ca629da4598917f4f283/pkg/apis/extensions/types.go#L189) it seems the behaviour I described above is not covered at all by whats described there and the GCP ingress controller is the only one doing it right. My best guess on why its happening is that is an implementation detail of the underlying data plane (nginx/envoy) leaking through.

Also it seems that [ingress-nginx does support  the regex, although it requires an annotation for that.](https://github.com/kubernetes/ingress-nginx/blob/c3eeaca97288ae6e03a68ed04697c454d9f2d024/docs/user-guide/ingress-path-matching.md#regular-expression-support)

I'll open an issue on Contour regarding the regex.

My take on fixing the docs here would be to revert this change and also add the `ingress-nginx` regex annotation.

One thing that confuses me right now thought: Is just `*` without a leading matcher something that should match everything? Isn't it just a quantifier?